testsupport/spaceprovisionerconfig/spaceprovisionerconfig.go

```diff
@@ -20,3 +23,59 @@ func CreateSpaceProvisionerConfig(t *testing.T, await *wait.Awaitility, opts ...
 
 	return spc
 }
+
+func UpdateForCluster(t *testing.T, await *wait.Awaitility, referencedClusterName string, opts ...testSpc.CreateOption) {
+	t.Helper()
+	spcs, err := getAllSpcs(t, await)
+	require.NoError(t, err)
+
+	idx := findIndexOfFirstSpaceProvisionerConfigReferencingCluster(spcs, referencedClusterName)
+	require.GreaterOrEqual(t, idx, 0, "could not find SpaceProvisionerConfig referencing the required cluster: %s", referencedClusterName)
+
+	spc := &spcs[idx]
+
+	originalSpc := spc.DeepCopy()
```

Comment:
if you rename and modify the function `findIndexOfFirstSpaceProvisionerConfigReferencingCluster` to:
```go
func findSPCFor(spcs []toolchainv1alpha1.SpaceProvisionerConfig, clusterName string) (*toolchainv1alpha1.SpaceProvisionerConfig, error) {
	for _, spc := range spcs {
		if spc.Spec.ToolchainCluster == clusterName {
			return &spc, nil
		}
	}
	return nil, fmt.Errorf("could not find SpaceProvisionerConfig referencing the required cluster: %s", clusterName)
}
```
then you could call it here: 
```suggestion
	spc, err := findSPCFor(spcs, referencedClusterName)
	require.NoError(t, err)

	originalSpc := spc.DeepCopy()
```
which seems to be a bit easier to read (at least for me)